### ISSUE TITLE
Inconsistent terminology in error message when "Name" and "Name on ID" match

### ISSUE THREAD

**Issue by SkylerRingtail** [MidwestFurryFandom/rams#98]
The error message for when "Name" and "Name on ID" match refers to a "preferred name," which is terminology not in use on the page itself. The terminology does appear in the name explanation pop-up when clicking the "What does this mean?" link. 

Additionally, the error message refers to a "legal name," which is also terminology not in use on the page itself.

**Suggested fix**: Rename the "Name" field label to "Preferred Name", and the "Name on ID" field to "Legal Name." Put "Legal Name" first as a required field, then add an optional field after for "Preferred Name." This will also eliminate the  "Above name is exactly what appears on my ID" checkbox and simplify the form slightly, resolving #99.

![mff preferred name](https://cloud.githubusercontent.com/assets/28108495/25568259/713a47a0-2dc4-11e7-8020-0b01cb11178f.png)

**Comment by kitsuta** [comment]
There's a pretty long history behind the way the naming schema was set up, and while it's not set in stone or anything, we explicitly used to have "Legal Name" as the required (and only) field but consistently had people fill in their preferred name because they wouldn't think about the fact that, e.g., "Jon" and "Jonathan" are different names. Only the people to whom this distinction matters the most (people with radically different legal names than the ones they go by) would enter their legal name.

So we still had difficulty checking in a lot of people, while singling out people whose old/dead names were not necessarily a comfortable subject. It was not an... ideal situation. The idea behind the checkbox is to force everyone to think, "no really, are you SURE that's the name on your photo ID?" which has the (hoped for) practical benefit of actually getting us the right data.

This is odd, though, because I was sure I removed this specific validation (it was, in fact, not good UX). The code might not have been updated on the staging server. :/
